src/katgpucbf/xbgpu/engine.py

```diff
@@ -860,28 +841,14 @@ def free_rx_item(self, item: RxQueueItem) -> None:
         if item.refcount == 0:
             # All Pipelines are done with this item
             self._active_in_sem.release()
-            # Need to wait for any outstanding events before recycling
-            self.add_service_task(
-                asyncio.create_task(
-                    self._push_recv_chunk(item),
-                    name="Receive Chunk Recycle Task",
-                )
-            )
+            # TODO: Don't recycle the chunk in the reset of the Item itself
+            # Need to separate the concerns
```

Comment:
Is there still something TODO?

Reply:
As far as I remember, this solution of `recycle()` just before/alongside the `item.reset()` isn't the most desirable?
* It was around the mention of "perhaps the Chunk will need to be reused before the RxQueueItem is done with it"
* Basically, the Chunk doesn't 'belong to' the RxQueueItem, so it should be "allowed to recycle" independently from the resetting of the item.

If that makes sense... Either way, perhaps not a great concern for right now?

Reply:
I just didn't want the call to `recycle` to be *inside* `reset`. I'm happy enough with the way things are.

We are delaying the release of chunks somewhat, so I would suggest running a test correlator at S-band rate before merging just to make sure this isn't going to cause any performance problems.